src/note.rs

```diff
@@ -0,0 +1,394 @@
+use kelvin::{ByteHash, Content, Sink, Source};
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::io::{self, Read, Write};
+
+use dusk_pki::jubjub_decode;
+use dusk_pki::Ownable;
+use dusk_pki::{PublicSpendKey, SecretSpendKey, StealthAddress, ViewKey};
+
+use dusk_plonk::jubjub::{dhke, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED};
+use poseidon252::cipher::PoseidonCipher;
+use poseidon252::sponge::sponge::sponge_hash;
+use rand::Rng;
+
+use crate::{
+    chunk_of, BlsScalar, Error, JubJubAffine, JubJubExtended, JubJubScalar,
+};
+
+const ENCRYPTED_DATA_SIZE: usize = 96;
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum NoteType {
+    Transparent = 0,
+    Obfuscated = 1,
+}
+
+impl TryFrom<u8> for NoteType {
+    type Error = Error;
+
+    fn try_from(note_type: u8) -> Result<Self, Self::Error> {
+        match note_type {
+            0 => Ok(NoteType::Transparent),
+            1 => Ok(NoteType::Obfuscated),
+            n => Err(Error::InvalidNoteType(n)),
+        }
+    }
+}
+
+impl TryFrom<i32> for NoteType {
+    type Error = Error;
+
+    fn try_from(note_type: i32) -> Result<Self, Self::Error> {
+        (note_type as u8).try_into()
+    }
+}
+
+/// A note that does not encrypt its value
+#[derive(Clone, Copy)]
+pub struct Note {
+    note_type: NoteType,
+    value_commitment: JubJubExtended,
+    nonce: JubJubScalar,
+    stealth_address: StealthAddress,
+    pos: u64,
+    encrypted_data: [u8; ENCRYPTED_DATA_SIZE],
+}
+
+impl PartialEq for Note {
+    fn eq(&self, other: &Self) -> bool {
+        self.hash() == other.hash()
+    }
+}
+
+impl Eq for Note {}
+
+impl Default for Note {
+    fn default() -> Self {
+        Note::new(NoteType::Transparent, &PublicSpendKey::default(), 0)
+    }
+}
+
+// TODO: Remove once we have encrypted data type
+impl std::fmt::Debug for Note {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Note")
+            .field("note_type", &self.note_type)
+            .field("value_commitment", &self.value_commitment)
+            .field("nonce", &self.nonce)
+            .field("stealth_address", &self.stealth_address)
+            .field("pos", &self.pos)
+            .field(
+                "encrypted_data",
+                &format_args!("{:?}", &self.encrypted_data),
+            )
+            .finish()
+    }
+}
```

Comment:
We have it. So this should be removed or the comment should not be there.

Reply:
No, we don't.
Basically we need to remove that once we have the "proper" encrypted data type (see comment above) and we don't have it yet – since it would be better waiting, otherwise we've 50% chance to re-write the wrong part of the code.

Reply:
kk. Fair enough